### PR TITLE
[PEPC-Boost][12] Custom devnet state interference checks tests

### PR DESCRIPTION
**Pull request by bharath-123**
## 📝 Summary

Tests for state interference checks

### CODE REVIEW

File: services/api/pepc_boost_test.go [hunk 1]
```diff
@@ -113,19 +117,180 @@ func assertTobTxs(t *testing.T, backend *testBackend, slot uint64, parentHash st
 
 }
 
-// TODO - this test will keep evolving as we expand the state interference checks
-func TestCheckTxAndSenderValidity(t *testing.T) {
+func GetCustomDevnetTracingRelatedTestData(t *testing.T) (*gethtypes.Transaction, *common.CallTrace, *gethtypes.Transaction, *common.CallTrace) {
```
Comment (michaelneuder) [inline services/api/pepc_boost_test.go:120]: t.Helper()

File: services/api/pepc_boost_test.go [hunk 2]
```diff
@@ -113,19 +117,180 @@ func assertTobTxs(t *testing.T, backend *testBackend, slot uint64, parentHash st
...
+	for _, c := range cases {
+		t.Run(c.description, func(t *testing.T) {
+			// Payload attributes
+			parentHash, feeRec, withdrawalsRoot, prevRandao, proposerPubkey, headSlot := GetTestPayloadAttributes(t)
+
+			prepareBackend(t, backend, headSlot, parentHash, feeRec, withdrawalsRoot, prevRandao, proposerPubkey, c.network)
+
+			res, err := backend.relay.IsTxWEthDaiSwap(c.callTraces)
+			if c.requiredError != "" {
+				require.Contains(t, err.Error(), c.requiredError)
```
Comment (michaelneuder) [inline services/api/pepc_boost_test.go:199]: can return in if and then move next to outer scope

File: services/api/pepc_boost_test.go [hunk 3]
```diff
@@ -113,19 +117,180 @@ func assertTobTxs(t *testing.T, backend *testBackend, slot uint64, parentHash st
...
+			requiredError:  "",
+		},
+	}
+
+	for _, c := range cases {
+		t.Run(c.description, func(t *testing.T) {
+			res, err := backend.relay.IsTraceToWEthDaiPair(c.callTrace)
+			if c.requiredError != "" {
+				require.Contains(t, err.Error(), c.requiredError)
+			} else {
```
Comment (michaelneuder) [inline services/api/pepc_boost_test.go:271]: same as above

**Review by michaelneuder — APPROVED** [review]
lgtm, starting to get a bit better feeling of how it works. nice use of the static files for txn examples

File: services/api/pepc_boost_test.go (hunk 1 above)
Comment (bharath-123) [reply services/api/pepc_boost_test.go:120]: done

File: services/api/pepc_boost_test.go (hunk 2 above)
Comment (bharath-123) [reply services/api/pepc_boost_test.go:199]: yeah yeah! will refactor all at once!

**Pull request merged**